great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -3063,17 +3063,25 @@ def _instantiate_datasource_from_config_and_update_project_config(
         Raises:
             DatasourceInitializationError
         """
-        substituted_config = self._perform_substitutions_on_datasource_config(config)
+        # Note that the call to `DatasourcStore.set` may alter the config object's state
+        # As such, we invoke it at the top of our function so any changes are reflected downstream
+        if save_changes:
+            config = self._datasource_store.set(key=None, value=config)  # type: ignore[attr-defined]
```

Comment:
Because we moved this line, our Cloud tests failed - we normally hydrated our config with `id` but didn't take that into consideration during our refactor.